lib/sda/treenode.go

```diff
@@ -0,0 +1,507 @@
+package sda
+
+import (
+	"errors"
+	"fmt"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/crypto/abstract"
+	"reflect"
+	"sync"
+)
+
+type TreeNodeInstance struct {
+	overlay *Overlay
+	token   *Token
+	// cache for the TreeNode this Node is representing
+	treeNode *TreeNode
+	// channels holds all channels available for the different message-types
+	channels map[network.MessageTypeID]interface{}
+	// registered handler-functions for that protocol
+	handlers map[network.MessageTypeID]interface{}
+	// flags for messages - only one channel/handler possible
+	messageTypeFlags map[network.MessageTypeID]uint32
```

Comment:
Just would like to have your input on whether you think we'll have other flags one day or not.